### PR TITLE
Install injection libraries with agent install script

### PR DESCRIPTION
**Pull request by randomanderson**
This PR adds the ability to install autoinstrumentation libraries after setting environment variables.

| Variable name | Description |
| ---------------|------------|
| DD_APM_INSTRUMENTATION_ENABLED | Install host and docker injection unless overridden with a more specific env variable |
| DD_APM_HOST_INJECTION_ENABLED |  Installs or prevents installation of host injection (via `LD_PRELOAD`) |
| DD_APM_DOCKER_INJECTION_ENABLED | Installs or prevents installation of docker injection ( via `daemon.json` ) |
| DD_APM_LIBRARIES | Space separated list of which languages to install support for. Valid values are `js`, `java`, `python`, `dotnet`, `all`.  Defaults to `all` if host or docker injection is enabled | 
| DD_NO_AGENT_INSTALL | If set, don't install the datadog agent. This is useful for the container injection only case. The check is for the existence of the variable, not any particular value. This mirrors DD_INSTALL_ONLY |

* The script does not verify that all combinations of new and existing variables make sense together (for instance `DD_NO_AGENT_INSTALL` and `DD_AGENT_MINOR_VERSION`)
* I removed the `Installing the $nice_flavor package` log statement because it is redundant with `Installing package(s) ...` line that follows and is misleading in some cases.

### Testing
I added few basic checks to the test suite. Only host injection could be tested because of the docker-in-docker issue.

### Additional Info:
* this PR depends on some fixes to the auto inject packages (https://github.com/DataDog/auto_inject/pull/57, https://github.com/DataDog/auto_inject/pull/55, https://github.com/DataDog/auto_inject/pull/54) being released

### CODE REVIEW

**Review by bkabrda — CHANGES_REQUESTED** [review]
Hi :wave: thanks a lot for sending the PR. It looks really good, I just left couple minor comments inline to get addressed and then I can approve. Thanks!

File: install_script.sh.template [hunk 1]
```diff
@@ -324,6 +329,54 @@ if [ -n "$DD_FIPS_MODE" ]; then
...
+fi
+
+docker_injection_enabled=
+if [[ "$DD_APM_DOCKER_INJECTION_ENABLED" == "true" || ( -z "$DD_APM_DOCKER_INJECTION_ENABLED" && "$DD_APM_INSTRUMENTATION_ENABLED" == "true") ]] ; then
+  docker_injection_enabled="true"
+  apm_libraries=("datadog-apm-inject")
+fi
+
+if [ -n "$DD_APM_LIBRARIES" ]; then
+  read -r -a lib_array <<< "$DD_APM_LIBRARIES"
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply install_script.sh.template:347]: For the datadog agent, we specifically use space separated lists: https://docs.datadoghq.com/agent/guide/environment-variables/#general-use

File: install_script.sh.template [hunk 2]
```diff
@@ -324,6 +329,54 @@ if [ -n "$DD_FIPS_MODE" ]; then
...
+        apm_libraries+=("datadog-apm-library-js")
+        ;;
+      python)
+        apm_libraries+=("datadog-apm-library-python")
+        ;;
+      dotnet)
+        apm_libraries+=("datadog-apm-library-dotnet")
+        ;;
+      *)
+        ERROR_MESSAGE="Unknown apm library: $lib. Must be one of: all, java, js, python, dotnet"
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply install_script.sh.template:367]: At most there is 1 more language to add: `ruby`. Go and native won't support runtime autoinstrumentation. The alternative is appending the given value to `datadog-apm-library-` and not giving an error until the package is not found. That's a worse user experience

File: install_script.sh.template (hunk 1 above)
[the remark this answers is not in the document]
Comment (bkabrda) [reply install_script.sh.template:347]: I didn't even know that. Thanks for the link, let's keep using spaces then.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -83,6 +83,20 @@ test:
     - ./test/dockertest.sh --image "registry.ddbuild.io/${IMAGE}" --script install_script_agent6.sh --minor-version "${MINOR_VERSION}" --expected-minor-version "${EXPECTED_MINOR_VERSION}" --flavor "${FLAVOR}"
     - ./test/dockertest.sh --image "registry.ddbuild.io/${IMAGE}" --script install_script_agent7.sh --minor-version "${MINOR_VERSION}" --expected-minor-version "${EXPECTED_MINOR_VERSION}" --flavor "${FLAVOR}"
 
+test-apm-injection:
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply .gitlab-ci.yml:86]: added

File: install_script.sh.template [hunk 4]
```diff
@@ -324,6 +329,54 @@ if [ -n "$DD_FIPS_MODE" ]; then
   fips_mode=$DD_FIPS_MODE
 fi
 
+declare -a apm_libraries
+
+host_injection_enabled=
```
[the remark this answers is not in the document]
Comment (randomanderson) [reply install_script.sh.template:334]: added a check

[the remark this answers is not in the document]
Comment (randomanderson) [reply install_script.sh.template:334]: done

**Comment by randomanderson** [comment]
closing in favor of a new PR that's not from a fork

**Pull request closed without merging**